pandas/core/generic.py

```diff
@@ -6815,16 +6815,21 @@ def mask(self, cond, other=np.nan, inplace=False, axis=None, level=None,
                           errors=errors)
 
     _shared_docs['shift'] = ("""
-        Shift index by desired number of periods with an optional time freq
+        Shift index by desired number of periods with an optional time freq.
+
+        When freq is not passed, shift the index without realign the data.
+        If freq is passed (in this case, the index must be date or datetime),
```

Comment:
What happens if `freq` is passed and the index is not datetime-like? Perhaps this should have a Raises section?

Reply:
Where is the docs to "Raises section"? I could not find it in https://python-sprints.github.io/pandas/guide/pandas_docstring.html

Reply:
It would look something like:

```
Returns
--------
same_class : yada yada

Raises
------
ValueError : something went terribly horribly wrong
```